rdflib/compare.py

```diff
@@ -438,42 +429,19 @@ def _traces(
             new_color = self._individuate(color_copy, candidate)
             coloring_copy.append(new_color)
             refined_coloring = self._refine(coloring_copy, [new_color])
-            color_score = tuple([c.key() for c in refined_coloring])
-            experimental = self._experimental_path(coloring_copy)
-            experimental_score = set([c.key() for c in experimental])
-            if last_coloring:
-                generator = self._create_generator(  # type: ignore[unreachable]
-                    [last_coloring, experimental], generator
-                )
-            last_coloring = experimental
-            if best_score is None or best_score < color_score:  # type: ignore[unreachable]
-                best = [refined_coloring]
+            color_score = tuple(c.key() for c in refined_coloring)
+
+            if best_score is None or best_score < color_score:
```

Comment:
The right-hand side of the `or` is never evaluated as `best_score` is always true here. Can you please review this.